File: shared/src/main/java/com/runtracker/shared/data/model/Gamification.kt
package com.runtracker.shared.data.model

import androidx.room.Entity
import androidx.room.PrimaryKey
import androidx.room.TypeConverters
import com.runtracker.shared.data.db.Converters

/**
 * User's gamification profile tracking XP, level, and streaks
 */
@Entity(tableName = "user_gamification")
@TypeConverters(Converters::class)
data class UserGamification(
    @PrimaryKey
    val id: Long = 1, // Single row for user
    val totalXp: Long = 0,
    val currentLevel: Int = 1,
    val currentStreak: Int = 0,
    val longestStreak: Int = 0,
    val lastActivityDate: Long? = null,
    val streakProtectionAvailable: Boolean = true,
    val totalWorkoutsCompleted: Int = 0,
    val totalDistanceMeters: Double = 0.0,
    val totalDurationMinutes: Int = 0,
    val weeklyWorkouts: Int = 0,
    val weekStartDate: Long = 0,
    val dailyGoalMinutes: Int = 30,
    val dailyGoalCompleted: Boolean = false,
    val weeklyGoalDays: Int = 4,
    val weeklyGoalProgress: Int = 0
) {
    val levelProgress: Float
        get() {
            val xpForCurrentLevel = xpRequiredForLevel(currentLevel)
            val xpForNextLevel = xpRequiredForLevel(currentLevel + 1)
            val xpInCurrentLevel = totalXp - xpForCurrentLevel
            val xpNeededForNext = xpForNextLevel - xpForCurrentLevel
            return (xpInCurrentLevel.toFloat() / xpNeededForNext).coerceIn(0f, 1f)
        }
    
    val xpToNextLevel: Long
        get() = xpRequiredForLevel(currentLevel + 1) - totalXp
    
    companion object {
        fun xpRequiredForLevel(level: Int): Long {
            // Exponential curve: each level requires more XP
            return when {
                level <= 1 -> 0
                else -> (100 * level * level * 0.5).toLong()
            }
        }
        
        fun levelForXp(xp: Long): Int {
            var level = 1
            while (xpRequiredForLevel(level + 1) <= xp) {
                level++
            }
            return level
        }
    }
}

/**
 * Achievement/Badge definition
 */
@Entity(tableName = "achievements")
data class Achievement(
    @PrimaryKey
    val id: String,
    val name: String,
    val description: String,
    val iconName: String,
    val category: AchievementCategory,
    val requirement: Int, // The threshold to unlock
    val xpReward: Int = 50,
    val isSecret: Boolean = false
)

/**
 * User's earned achievements
 */
@Entity(tableName = "user_achievements")
data class UserAchievement(
    @PrimaryKey(autoGenerate = true)
    val id: Long = 0,
    val achievementId: String,
    val unlockedAt: Long = System.currentTimeMillis(),
    val progress: Int = 0, // Current progress toward achievement
    val isUnlocked: Boolean = false,
    val notificationShown: Boolean = false
)

enum class AchievementCategory {
    DISTANCE,       // Total distance milestones
    WORKOUTS,       // Number of workouts
    STREAKS,        // Streak achievements
    SPEED,          // Speed/pace achievements
    CONSISTENCY,    // Weekly/monthly consistency
    SOCIAL,         // Social interactions
    EXPLORATION,    // New routes, locations
    PERSONAL_BEST,  // PB achievements
    SPECIAL         // Special/seasonal achievements
}

/**
 * Daily activity ring progress
 */
@Entity(tableName = "daily_rings")
data class DailyRings(
    @PrimaryKey
    val date: Long, // Start of day timestamp
    val moveMinutes: Int = 0,
    val moveGoal: Int = 30,
    val exerciseMinutes: Int = 0,
    val exerciseGoal: Int = 30,
    val standHours: Int = 0,
    val standGoal: Int = 12,
    val caloriesBurned: Int = 0,
    val caloriesGoal: Int = 500,
    val distanceMeters: Double = 0.0,
    val distanceGoal: Double = 5000.0,
    val allRingsClosed: Boolean = false
) {
    val moveProgress: Float get() = (moveMinutes.toFloat() / moveGoal).coerceIn(0f, 1f)
    val exerciseProgress: Float get() = (exerciseMinutes.toFloat() / exerciseGoal).coerceIn(0f, 1f)
    val standProgress: Float get() = (standHours.toFloat() / standGoal).coerceIn(0f, 1f)
    val caloriesProgress: Float get() = (caloriesBurned.toFloat() / caloriesGoal).coerceIn(0f, 1f)
    val distanceProgress: Float get() = (distanceMeters / distanceGoal).toFloat().coerceIn(0f, 1f)
}

/**
 * XP transaction log
 */
@Entity(tableName = "xp_transactions")
data class XpTransaction(
    @PrimaryKey(autoGenerate = true)
    val id: Long = 0,
    val amount: Int,
    val reason: XpReason,
    val relatedId: Long? = null, // Run ID, achievement ID, etc.
    val timestamp: Long = System.currentTimeMillis()
)

enum class XpReason {
    WORKOUT_COMPLETED,
    STREAK_BONUS,
    ACHIEVEMENT_UNLOCKED,
    DAILY_GOAL_COMPLETED,
    WEEKLY_GOAL_COMPLETED,
    PERSONAL_BEST,
    CHALLENGE_COMPLETED,
    FIRST_WORKOUT,
    LEVEL_UP_BONUS
}

/**
 * Pre-defined achievements
 */
object DefaultAchievements {
    
    fun getAll(): List<Achievement> = distanceAchievements + workoutAchievements + 
            streakAchievements + speedAchievements + consistencyAchievements + pbAchievements
    
    val distanceAchievements = listOf(
        Achievement("dist_1k", "First Kilometer", "Run your first kilometer", "directions_run", AchievementCategory.DISTANCE, 1000, 25),
        Achievement("dist_5k", "5K Club", "Run a total of 5 kilometers", "directions_run", AchievementCategory.DISTANCE, 5000, 50),
        Achievement("dist_10k", "Double Digits", "Run a total of 10 kilometers", "directions_run", AchievementCategory.DISTANCE, 10000, 75),
        Achievement("dist_42k", "Marathon Distance", "Run a total of 42.195 kilometers", "emoji_events", AchievementCategory.DISTANCE, 42195, 150),
        Achievement("dist_100k", "Century Club", "Run a total of 100 kilometers", "military_tech", AchievementCategory.DISTANCE, 100000, 200),
        Achievement("dist_500k", "Road Warrior", "Run a total of 500 kilometers", "military_tech", AchievementCategory.DISTANCE, 500000, 500),
        Achievement("dist_1000k", "Thousand K Legend", "Run a total of 1000 kilometers", "workspace_premium", AchievementCategory.DISTANCE, 1000000, 1000)
    )
    
    val workoutAchievements = listOf(
        Achievement("workout_1", "First Steps", "Complete your first workout", "fitness_center", AchievementCategory.WORKOUTS, 1, 50),
        Achievement("workout_10", "Getting Started", "Complete 10 workouts", "fitness_center", AchievementCategory.WORKOUTS, 10, 100),
        Achievement("workout_25", "Quarter Century", "Complete 25 workouts", "fitness_center", AchievementCategory.WORKOUTS, 25, 150),
        Achievement("workout_50", "Halfway Hero", "Complete 50 workouts", "fitness_center", AchievementCategory.WORKOUTS, 50, 250),
        Achievement("workout_100", "Century Runner", "Complete 100 workouts", "emoji_events", AchievementCategory.WORKOUTS, 100, 500),
        Achievement("workout_250", "Dedicated Athlete", "Complete 250 workouts", "military_tech", AchievementCategory.WORKOUTS, 250, 750),
        Achievement("workout_500", "Elite Runner", "Complete 500 workouts", "workspace_premium", AchievementCategory.WORKOUTS, 500, 1000)
    )
    
    val streakAchievements = listOf(
        Achievement("streak_3", "Hat Trick", "Maintain a 3-day streak", "local_fire_department", AchievementCategory.STREAKS, 3, 30),
        Achievement("streak_7", "Week Warrior", "Maintain a 7-day streak", "local_fire_department", AchievementCategory.STREAKS, 7, 75),
        Achievement("streak_14", "Fortnight Fighter", "Maintain a 14-day streak", "local_fire_department", AchievementCategory.STREAKS, 14, 150),
        Achievement("streak_30", "Monthly Master", "Maintain a 30-day streak", "whatshot", AchievementCategory.STREAKS, 30, 300),
        Achievement("streak_60", "Two Month Titan", "Maintain a 60-day streak", "whatshot", AchievementCategory.STREAKS, 60, 500),
        Achievement("streak_100", "Century Streak", "Maintain a 100-day streak", "workspace_premium", AchievementCategory.STREAKS, 100, 1000),
        Achievement("streak_365", "Year of Dedication", "Maintain a 365-day streak", "workspace_premium", AchievementCategory.STREAKS, 365, 5000, isSecret = true)
    )
    
    val speedAchievements = listOf(
        Achievement("pace_6min", "Sub-6 Minute Mile", "Run a kilometer under 6:00 pace", "speed", AchievementCategory.SPEED, 360, 100),
        Achievement("pace_5min", "Sub-5 Minute Mile", "Run a kilometer under 5:00 pace", "speed", AchievementCategory.SPEED, 300, 200),
        Achievement("pace_4min", "Sub-4 Minute Mile", "Run a kilometer under 4:00 pace", "speed", AchievementCategory.SPEED, 240, 500, isSecret = true)
    )
    
    val consistencyAchievements = listOf(
        Achievement("weekly_3", "Tri-Weekly", "Work out 3 times in a week", "date_range", AchievementCategory.CONSISTENCY, 3, 50),
        Achievement("weekly_5", "Five-a-Week", "Work out 5 times in a week", "date_range", AchievementCategory.CONSISTENCY, 5, 100),
        Achievement("weekly_7", "Perfect Week", "Work out every day for a week", "date_range", AchievementCategory.CONSISTENCY, 7, 200),
        Achievement("monthly_20", "Monthly Dedication", "Complete 20 workouts in a month", "calendar_month", AchievementCategory.CONSISTENCY, 20, 300)
    )
    
    val pbAchievements = listOf(
        Achievement("pb_first", "Personal Best!", "Set your first personal best", "emoji_events", AchievementCategory.PERSONAL_BEST, 1, 100),
        Achievement("pb_5", "Record Breaker", "Set 5 personal bests", "emoji_events", AchievementCategory.PERSONAL_BEST, 5, 200),
        Achievement("pb_10", "PB Machine", "Set 10 personal bests", "emoji_events", AchievementCategory.PERSONAL_BEST, 10, 400)
    )
}

/**
 * XP rewards for different activities
 */
object XpRewards {
    const val WORKOUT_BASE = 50
    const val PER_KILOMETER = 10
    const val PER_MINUTE = 2
    const val STREAK_BONUS_PER_DAY = 5
    const val DAILY_GOAL_BONUS = 25
    const val WEEKLY_GOAL_BONUS = 100
    const val PERSONAL_BEST = 75
    const val FIRST_WORKOUT = 100
    
    fun calculateWorkoutXp(distanceMeters: Double, durationMinutes: Int, currentStreak: Int): Int {
        val distanceXp = (distanceMeters / 1000 * PER_KILOMETER).toInt()
        val durationXp = durationMinutes * PER_MINUTE
        val streakBonus = currentStreak * STREAK_BONUS_PER_DAY
        return WORKOUT_BASE + distanceXp + durationXp + streakBonus
    }
}

File: iosApp/GoSteady/Sources/Models/Gamification.swift
import Foundation
import SwiftData

@Model
final class UserGamification {
    @Attribute(.unique) var id: Int
    var totalXp: Int64
    var currentLevel: Int
    var currentStreak: Int
    var longestStreak: Int
    var lastActivityDate: Date?
    var streakProtectionAvailable: Bool
    var totalWorkoutsCompleted: Int
    var totalDistanceMeters: Double
    var totalDurationMinutes: Int
    var weeklyWorkouts: Int
    var weekStartDate: Date
    var dailyGoalMinutes: Int
    var dailyGoalCompleted: Bool
    var weeklyGoalDays: Int
    var weeklyGoalProgress: Int

    var levelProgress: Float {
        let xpForCurrent = UserGamification.xpRequiredForLevel(currentLevel)
        let xpForNext = UserGamification.xpRequiredForLevel(currentLevel + 1)
        let xpInCurrent = totalXp - xpForCurrent
        let xpNeeded = xpForNext - xpForCurrent
        guard xpNeeded > 0 else { return 0 }
        return min(max(Float(xpInCurrent) / Float(xpNeeded), 0), 1)
    }

    var xpToNextLevel: Int64 {
        UserGamification.xpRequiredForLevel(currentLevel + 1) - totalXp
    }

    static func xpRequiredForLevel(_ level: Int) -> Int64 {
        guard level > 1 else { return 0 }
        return Int64(100.0 * Double(level * level) * 0.5)
    }

    static func levelForXp(_ xp: Int64) -> Int {
        var level = 1
        while xpRequiredForLevel(level + 1) <= xp {
            level += 1
        }
        return level
    }

    init(
        id: Int = 1,
        totalXp: Int64 = 0,
        currentLevel: Int = 1,
        currentStreak: Int = 0,
        longestStreak: Int = 0,
        lastActivityDate: Date? = nil,
        streakProtectionAvailable: Bool = true,
        totalWorkoutsCompleted: Int = 0,
        totalDistanceMeters: Double = 0.0,
        totalDurationMinutes: Int = 0,
        weeklyWorkouts: Int = 0,
        weekStartDate: Date = Date(),
        dailyGoalMinutes: Int = 30,
        dailyGoalCompleted: Bool = false,
        weeklyGoalDays: Int = 4,
        weeklyGoalProgress: Int = 0
    ) {
        self.id = id
        self.totalXp = totalXp
        self.currentLevel = currentLevel
        self.currentStreak = currentStreak
        self.longestStreak = longestStreak
        self.lastActivityDate = lastActivityDate
        self.streakProtectionAvailable = streakProtectionAvailable
        self.totalWorkoutsCompleted = totalWorkoutsCompleted
        self.totalDistanceMeters = totalDistanceMeters
        self.totalDurationMinutes = totalDurationMinutes
        self.weeklyWorkouts = weeklyWorkouts
        self.weekStartDate = weekStartDate
        self.dailyGoalMinutes = dailyGoalMinutes
        self.dailyGoalCompleted = dailyGoalCompleted
        self.weeklyGoalDays = weeklyGoalDays
        self.weeklyGoalProgress = weeklyGoalProgress
    }
}

@Model
final class Achievement {
    @Attribute(.unique) var id: String
    var name: String
    var achievementDescription: String
    var iconName: String
    var category: AchievementCategory
    var requirement: Int
    var xpReward: Int
    var isSecret: Bool

    init(
        id: String,
        name: String,
        achievementDescription: String,
        iconName: String,
        category: AchievementCategory,
        requirement: Int,
        xpReward: Int = 50,
        isSecret: Bool = false
    ) {
        self.id = id
        self.name = name
        self.achievementDescription = achievementDescription
        self.iconName = iconName
        self.category = category
        self.requirement = requirement
        self.xpReward = xpReward
        self.isSecret = isSecret
    }
}

@Model
final class UserAchievement {
    @Attribute(.unique) var id: UUID
    var achievementId: String
    var unlockedAt: Date
    var progress: Int
    var isUnlocked: Bool
    var notificationShown: Bool

    init(
        id: UUID = UUID(),
        achievementId: String,
        unlockedAt: Date = Date(),
        progress: Int = 0,
        isUnlocked: Bool = false,
        notificationShown: Bool = false
    ) {
        self.id = id
        self.achievementId = achievementId
        self.unlockedAt = unlockedAt
        self.progress = progress
        self.isUnlocked = isUnlocked
        self.notificationShown = notificationShown
    }
}

enum AchievementCategory: String, Codable, CaseIterable, Identifiable {
    case distance = "DISTANCE"
    case workouts = "WORKOUTS"
    case streaks = "STREAKS"
    case speed = "SPEED"
    case consistency = "CONSISTENCY"
    case social = "SOCIAL"
    case exploration = "EXPLORATION"
    case personalBest = "PERSONAL_BEST"
    case special = "SPECIAL"

    var id: String { rawValue }

    var displayName: String {
        switch self {
        case .distance: return "Distance"
        case .workouts: return "Workouts"
        case .streaks: return "Streaks"
        case .speed: return "Speed"
        case .consistency: return "Consistency"
        case .social: return "Social"
        case .exploration: return "Exploration"
        case .personalBest: return "Personal Best"
        case .special: return "Special"
        }
    }
}

@Model
final class DailyRings {
    @Attribute(.unique) var date: Date
    var moveMinutes: Int
    var moveGoal: Int
    var exerciseMinutes: Int
    var exerciseGoal: Int
    var standHours: Int
    var standGoal: Int
    var caloriesBurned: Int
    var caloriesGoal: Int
    var distanceMeters: Double
    var distanceGoal: Double
    var allRingsClosed: Bool

    var moveProgress: Float { min(Float(moveMinutes) / Float(moveGoal), 1.0) }
    var exerciseProgress: Float { min(Float(exerciseMinutes) / Float(exerciseGoal), 1.0) }
    var standProgress: Float { min(Float(standHours) / Float(standGoal), 1.0) }
    var caloriesProgress: Float { min(Float(caloriesBurned) / Float(caloriesGoal), 1.0) }
    var distanceProgress: Float { min(Float(distanceMeters / distanceGoal), 1.0) }

    init(
        date: Date,
        moveMinutes: Int = 0,
        moveGoal: Int = 30,
        exerciseMinutes: Int = 0,
        exerciseGoal: Int = 30,
        standHours: Int = 0,
        standGoal: Int = 12,
        caloriesBurned: Int = 0,
        caloriesGoal: Int = 500,
        distanceMeters: Double = 0.0,
        distanceGoal: Double = 5000.0,
        allRingsClosed: Bool = false
    ) {
        self.date = date
        self.moveMinutes = moveMinutes
        self.moveGoal = moveGoal
        self.exerciseMinutes = exerciseMinutes
        self.exerciseGoal = exerciseGoal
        self.standHours = standHours
        self.standGoal = standGoal
        self.caloriesBurned = caloriesBurned
        self.caloriesGoal = caloriesGoal
        self.distanceMeters = distanceMeters
        self.distanceGoal = distanceGoal
        self.allRingsClosed = allRingsClosed
    }
}

@Model
final class XpTransaction {
    @Attribute(.unique) var id: UUID
    var amount: Int
    var reason: XpReason
    var relatedId: UUID?
    var timestamp: Date

    init(
        id: UUID = UUID(),
        amount: Int,
        reason: XpReason,
        relatedId: UUID? = nil,
        timestamp: Date = Date()
    ) {
        self.id = id
        self.amount = amount
        self.reason = reason
        self.relatedId = relatedId
        self.timestamp = timestamp
    }
}

enum XpReason: String, Codable, CaseIterable {
    case workoutCompleted = "WORKOUT_COMPLETED"
    case streakBonus = "STREAK_BONUS"
    case achievementUnlocked = "ACHIEVEMENT_UNLOCKED"
    case dailyGoalCompleted = "DAILY_GOAL_COMPLETED"
    case weeklyGoalCompleted = "WEEKLY_GOAL_COMPLETED"
    case personalBest = "PERSONAL_BEST"
    case challengeCompleted = "CHALLENGE_COMPLETED"
    case firstWorkout = "FIRST_WORKOUT"
    case levelUpBonus = "LEVEL_UP_BONUS"

    var displayName: String {
        switch self {
        case .workoutCompleted: return "Workout Completed"
        case .streakBonus: return "Streak Bonus"
        case .achievementUnlocked: return "Achievement Unlocked"
        case .dailyGoalCompleted: return "Daily Goal"
        case .weeklyGoalCompleted: return "Weekly Goal"
        case .personalBest: return "Personal Best"
        case .challengeCompleted: return "Challenge Completed"
        case .firstWorkout: return "First Workout"
        case .levelUpBonus: return "Level Up Bonus"
        }
    }
}

enum XpRewards {
    static let workoutBase = 50
    static let perKilometer = 10
    static let perMinute = 2
    static let streakBonusPerDay = 5
    static let dailyGoalBonus = 25
    static let weeklyGoalBonus = 100
    static let personalBest = 75
    static let firstWorkout = 100

    static func calculateWorkoutXp(distanceMeters: Double, durationMinutes: Int, currentStreak: Int) -> Int {
        let distanceXp = Int(distanceMeters / 1000 * Double(perKilometer))
        let durationXp = durationMinutes * perMinute
        let streakBonus = currentStreak * streakBonusPerDay
        return workoutBase + distanceXp + durationXp + streakBonus
    }
}

enum DefaultAchievements {
    static let all: [Achievement] = distance + workouts + streaks + speed + consistency + personalBests

    static let distance: [Achievement] = [
        Achievement(id: "dist_1k", name: "First Kilometer", achievementDescription: "Run your first kilometer", iconName: "figure.run", category: .distance, requirement: 1000, xpReward: 25),
        Achievement(id: "dist_5k", name: "5K Club", achievementDescription: "Run a total of 5 kilometers", iconName: "figure.run", category: .distance, requirement: 5000, xpReward: 50),
        Achievement(id: "dist_10k", name: "Double Digits", achievementDescription: "Run a total of 10 kilometers", iconName: "figure.run", category: .distance, requirement: 10000, xpReward: 75),
        Achievement(id: "dist_42k", name: "Marathon Distance", achievementDescription: "Run a total of 42.195 kilometers", iconName: "trophy", category: .distance, requirement: 42195, xpReward: 150),
        Achievement(id: "dist_100k", name: "Century Club", achievementDescription: "Run a total of 100 kilometers", iconName: "medal", category: .distance, requirement: 100000, xpReward: 200),
        Achievement(id: "dist_500k", name: "Road Warrior", achievementDescription: "Run a total of 500 kilometers", iconName: "medal", category: .distance, requirement: 500000, xpReward: 500),
        Achievement(id: "dist_1000k", name: "Thousand K Legend", achievementDescription: "Run a total of 1000 kilometers", iconName: "star.circle.fill", category: .distance, requirement: 1000000, xpReward: 1000),
    ]

    static let workouts: [Achievement] = [
        Achievement(id: "workout_1", name: "First Steps", achievementDescription: "Complete your first workout", iconName: "dumbbell", category: .workouts, requirement: 1, xpReward: 50),
        Achievement(id: "workout_10", name: "Getting Started", achievementDescription: "Complete 10 workouts", iconName: "dumbbell", category: .workouts, requirement: 10, xpReward: 100),
        Achievement(id: "workout_25", name: "Quarter Century", achievementDescription: "Complete 25 workouts", iconName: "dumbbell", category: .workouts, requirement: 25, xpReward: 150),
        Achievement(id: "workout_50", name: "Halfway Hero", achievementDescription: "Complete 50 workouts", iconName: "dumbbell", category: .workouts, requirement: 50, xpReward: 250),
        Achievement(id: "workout_100", name: "Century Runner", achievementDescription: "Complete 100 workouts", iconName: "trophy", category: .workouts, requirement: 100, xpReward: 500),
        Achievement(id: "workout_250", name: "Dedicated Athlete", achievementDescription: "Complete 250 workouts", iconName: "medal", category: .workouts, requirement: 250, xpReward: 750),
        Achievement(id: "workout_500", name: "Elite Runner", achievementDescription: "Complete 500 workouts", iconName: "star.circle.fill", category: .workouts, requirement: 500, xpReward: 1000),
    ]

    static let streaks: [Achievement] = [
        Achievement(id: "streak_3", name: "Hat Trick", achievementDescription: "Maintain a 3-day streak", iconName: "flame", category: .streaks, requirement: 3, xpReward: 30),
        Achievement(id: "streak_7", name: "Week Warrior", achievementDescription: "Maintain a 7-day streak", iconName: "flame", category: .streaks, requirement: 7, xpReward: 75),
        Achievement(id: "streak_14", name: "Fortnight Fighter", achievementDescription: "Maintain a 14-day streak", iconName: "flame", category: .streaks, requirement: 14, xpReward: 150),
        Achievement(id: "streak_30", name: "Monthly Master", achievementDescription: "Maintain a 30-day streak", iconName: "flame.fill", category: .streaks, requirement: 30, xpReward: 300),
        Achievement(id: "streak_100", name: "Century Streak", achievementDescription: "Maintain a 100-day streak", iconName: "star.circle.fill", category: .streaks, requirement: 100, xpReward: 1000),
        Achievement(id: "streak_365", name: "Year of Dedication", achievementDescription: "Maintain a 365-day streak", iconName: "star.circle.fill", category: .streaks, requirement: 365, xpReward: 5000, isSecret: true),
    ]

    static let speed: [Achievement] = [
        Achievement(id: "pace_6min", name: "Sub-6 Pace", achievementDescription: "Run a kilometer under 6:00 pace", iconName: "hare", category: .speed, requirement: 360, xpReward: 100),
        Achievement(id: "pace_5min", name: "Sub-5 Pace", achievementDescription: "Run a kilometer under 5:00 pace", iconName: "hare", category: .speed, requirement: 300, xpReward: 200),
        Achievement(id: "pace_4min", name: "Sub-4 Pace", achievementDescription: "Run a kilometer under 4:00 pace", iconName: "hare", category: .speed, requirement: 240, xpReward: 500, isSecret: true),
    ]

    static let consistency: [Achievement] = [
        Achievement(id: "weekly_3", name: "Tri-Weekly", achievementDescription: "Work out 3 times in a week", iconName: "calendar", category: .consistency, requirement: 3, xpReward: 50),
        Achievement(id: "weekly_5", name: "Five-a-Week", achievementDescription: "Work out 5 times in a week", iconName: "calendar", category: .consistency, requirement: 5, xpReward: 100),
        Achievement(id: "weekly_7", name: "Perfect Week", achievementDescription: "Work out every day for a week", iconName: "calendar", category: .consistency, requirement: 7, xpReward: 200),
        Achievement(id: "monthly_20", name: "Monthly Dedication", achievementDescription: "Complete 20 workouts in a month", iconName: "calendar.badge.checkmark", category: .consistency, requirement: 20, xpReward: 300),
    ]

    static let personalBests: [Achievement] = [
        Achievement(id: "pb_first", name: "Personal Best!", achievementDescription: "Set your first personal best", iconName: "trophy", category: .personalBest, requirement: 1, xpReward: 100),
        Achievement(id: "pb_5", name: "Record Breaker", achievementDescription: "Set 5 personal bests", iconName: "trophy", category: .personalBest, requirement: 5, xpReward: 200),
        Achievement(id: "pb_10", name: "PB Machine", achievementDescription: "Set 10 personal bests", iconName: "trophy", category: .personalBest, requirement: 10, xpReward: 400),
    ]
}
